### PR TITLE
Align on Transport GID and Liveliness GUID

### PR DESCRIPTION
**Pull request by ahcorde**
Related with https://github.com/ros2/rmw_zenoh/issues/290 and https://github.com/ros2/rmw_zenoh/issues/148

This PR makes `test_node_interfaces__node_graph` from rclcpp to pass.

### CODE REVIEW

File: rmw_zenoh_cpp/src/detail/graph_cache.cpp [hunk 1]
```diff
@@ -1208,7 +1208,26 @@ rmw_ret_t GraphCache::get_entities_info_by_topic(
...
+        liveliness::ConstEntityPtr entity;
+        if (entity_type == EntityType::Publisher) {
+          entity = (*topic_data->pubs_.begin());
+        } else {
+          entity = (*topic_data->subs_.begin());
+        }
+
+        if (entity) {
+          uint8_t gid[RMW_GID_STORAGE_SIZE];
+          entity->get_gid(gid);
```
Comment (Yadunund) [inline rmw_zenoh_cpp/src/detail/graph_cache.cpp:1220]: Should we iterate though all the `Entity` ptrs in the `EnittySet`?
```
const auto & entity_set = entity_type == EntityType::Publisher? topic_data->pubs_ : topic_data->subs_;
for (auto entity_it = entity_set.begin(); entity_it != entity_set.end(); ++entity_it) {
   uint8_t gid[RMW_GID_STORAGE_SIZE];
   entity->get_gid(gid);
    ret = rmw_topic_endpoint_info_set_gid(
        &endpoint_info,
        gid,
        RMW_GID_STORAGE_SIZE
     );
     if (RMW_RET_OK != ret) {
        return ret;
          }
}

```

File: rmw_zenoh_cpp/src/detail/liveliness_utils.hpp [hunk 2]
```diff
@@ -153,6 +153,12 @@ class Entity
   // This is named guid and not gid to remain distinct as it is not of type rmw_gid_t.
   std::size_t guid() const;
 
+  // Generate the 16byte GID from the keyexpr guid
+  // The size of guid is std::size_t. It can vary depending on the platform and architecture.
+  // On most 32-bit systems, it is usually 32 bits, while on 64-bit systems,
+  // it is typically 64 bits. It fit in the RMW_GID_STORAGE_SIZE.
+  void get_gid(uint8_t gid[RMW_GID_STORAGE_SIZE]) const;
```
Comment (Yadunund) [inline rmw_zenoh_cpp/src/detail/liveliness_utils.hpp:160]: Nit: To be consistent with naming of other getters.
```suggestion
  void gid(uint8_t gid[RMW_GID_STORAGE_SIZE]) const;
```

**Review by Yadunund — CHANGES_REQUESTED** [review]

File: rmw_zenoh_cpp/src/detail/graph_cache.cpp (hunk 1 above)
Comment (ahcorde) [reply rmw_zenoh_cpp/src/detail/graph_cache.cpp:1220]: If we iterate over the list, then it will override the gid value (if there is more than one).

**Comment by clalancette** [comment]
I'm a little later here, but please see https://github.com/ros2/rmw_zenoh/issues/290#issuecomment-2397193935 for some of my thoughts.

**Comment by clalancette** [comment]
Closing in favor of #298 .

**Pull request closed without merging**